### PR TITLE
EASY-2116: add language of files to DDM

### PR DESCRIPTION
**Pull request by stephanHogenboom**
Fixes EASY-2116: add language of files to DDM

#### When applied it will
*  add language of files to DDM

#### See also:
* https://github.com/DANS-KNAW/easy-ddm/pull/34

### CODE REVIEW

File: src/test/scala/nl.knaw.dans.easy.deposit/docs/DDMSpec.scala [hunk 1]
```diff
@@ -537,6 +537,10 @@ class DDMSpec extends TestSupportFixture with DdmBehavior {
           </boundedBy>
         </dcx-gml:spatial>
         <dcterms:license xsi:type="dcterms:URI">http://creativecommons.org/publicdomain/zero/1.0</dcterms:license>
+        <dcterms:language xsi:type="dcterms:ISO639-2">English</dcterms:language>
+        <dcterms:language xsi:type="dcterms:ISO639-2">Dutch</dcterms:language>
```
Comment (lindareijnhoudt) [inline src/test/scala/nl.knaw.dans.easy.deposit/docs/DDMSpec.scala]: these values are not in iso639-2, should be `nld` or `dut` 
```suggestion
        <dcterms:language xsi:type="dcterms:ISO639-2">nld</dcterms:language>
```

Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.deposit/docs/DDMSpec.scala]: @stephanHogenboom I'm not sure your changes are what @lindareijnhoudt intended. You're using `lang.value`, but shouldn't it be `lang.key`? See

https://github.com/DANS-KNAW/easy-deposit-api/blob/d023e0755d2a7dee41e52e587a15555b27236263/src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala#L72

In your changes you changed the example input data, but that isn't how to solve this. Could you test this together with `easy-deposit-ui` as well?

File: src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala [hunk 2]
```diff
@@ -69,19 +69,20 @@ object DDM extends SchemedXml with DebugEnhancedLogging {
         { dm.spatialPoints.withNonEmpty.map(point => <dcx-gml:spatial srsName={ point.srsName }>{ details(point) }</dcx-gml:spatial>) }
         { dm.spatialBoxes.withNonEmpty.map(point => <dcx-gml:spatial>{ details(point) }</dcx-gml:spatial>) }
         { dm.license.withNonEmpty.map(src => <dcterms:license xsi:type={ src.scheme.nonBlankOrNull }>{ src.value.nonBlankOrEmpty }</dcterms:license>) }
+        { dm.languagesOfFiles.withNonEmpty.flatMap(lang => lang.value.map(value => <dcterms:language xsi:type ={ lang.scheme.nonBlankOrNull  }>{ value }</dcterms:language>)) }
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala]: As mentioned in https://github.com/DANS-KNAW/easy-deposit-api/pull/158#discussion_r295351969, probably you should use `lang.key` here? Please discuss with @lindareijnhoudt first!

File: src/test/resources/manual-test/datasetmetadata-from-ui-all.json [hunk 3]
```diff
@@ -166,7 +166,7 @@
     {
       "scheme": "dcterms:ISO639-2",
       "key": "nld",
-      "value": "Dutch"
+      "value": "nld"
```
Comment (rvanheest) [inline src/test/resources/manual-test/datasetmetadata-from-ui-all.json]: This `value` should be `"Dutch"`, just as it is `"English"` above.

File: src/test/scala/nl.knaw.dans.easy.deposit/docs/DDMSpec.scala [hunk 4]
```diff
@@ -537,6 +537,10 @@ class DDMSpec extends TestSupportFixture with DdmBehavior {
           </boundedBy>
         </dcx-gml:spatial>
         <dcterms:license xsi:type="dcterms:URI">http://creativecommons.org/publicdomain/zero/1.0</dcterms:license>
+        <dcterms:language xsi:type="dcterms:ISO639-2">English</dcterms:language>
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/docs/DDMSpec.scala]: I expect `eng` here, just as @lindareijnhoudt pointed out on `nld` below, but it should be using the object's key, rather than the value.

File: src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala (hunk 2 above)
Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala]: So KeyLess values like "Goerees" or other non ISO639-2 values are lost? I'm quite lost here now frankly. Ill will discuss this with @lindareijnhoudt

File: src/test/resources/manual-test/datasetmetadata-from-ui-all.json [hunk 5]
```diff
@@ -166,7 +166,7 @@
     {
       "scheme": "dcterms:ISO639-2",
       "key": "nld",
-      "value": "Dutch"
+      "value": "DUTCH"
```
Comment (rvanheest) [inline src/test/resources/manual-test/datasetmetadata-from-ui-all.json]: Please revert this change
